### DISCUSSION TITLE
How can i convert quaternion to vector3 values ?

### DISCUSSION THREAD

**Discussion by suryanadiminti99** [sceneview/sceneform-android#132] [Q&A]
Hi Team, How can I convert my quaternion to vector3 values. 
below is the reference link for the online converter. But how can I implement the same in java? 
@ThomasGorisse 

[https://www.andre-gaschler.com/rotationconverter/](https://www.andre-gaschler.com/rotationconverter/)

**Comment by RGregat** [comment]
I took a look on the mentioned Repository for this Website and found this formula

```javascript
var axis = [0, 0, 0];
var angle = 2 * Math.acos(q.w);
if (1 - (q.w * q.w) < 0.000001)
{
    axis[0] = q.x;
    axis[1] = q.y;
    axis[2] = q.z;
}
else
{
    // http://www.euclideanspace.com/maths/geometry/rotations/conversions/quaternionToAngle/
    var s = Math.sqrt(1 - (q.w * q.w));
    axis[0] = q.x / s;
    axis[1] = q.y / s;
    axis[2] = q.z / s;
}
```

I guess that is what you want? If so here is the blueprint, the conversion to java shouldn't be to hard :)

**Reply by suryanadiminti99** [reply]
Thank you, I will try to integrate it with my code.
